packages/compass-collection/src/stores/context.tsx

```diff
@@ -4,11 +4,40 @@ import semver from 'semver';
 import { ErrorBoundary } from '@mongodb-js/compass-components';
 import { createLoggerAndTelemetry } from '@mongodb-js/compass-logging';
 import type { Document } from 'mongodb';
+import type { DataService } from 'mongodb-data-service';
+import ConnectionString from 'mongodb-connection-string-url';
 
 const { log, mongoLogId } = createLoggerAndTelemetry(
   'mongodb-compass:compass-collection:context'
 );
 
+// TODO: replace this with state coming from the right layer.
+// this kind of information should not be derived
+// from dataService as it operates on a lower level,
+// as a consequence here we have to remove the `appName` that only
+// the dataService should be using.
```

Comment:
Moved this comment too as I'm not sure if having this method in this store is any better than having it in export-to-language store